### PR TITLE
Fix WooPay phone number on blocks checkout

### PR DESCRIPTION
**Pull request by alefesouza**
Fixes 1610-gh-Automattic/woopay

#### Changes proposed in this Pull Request

This PR fixes a bug that happens when signing up to WooPay with blocks checkout when the user filled out a phone number without country code.

#### Testing instructions

* Add a product to cart.
* Go to blocks checkout.
* Type an e-mail address that doesn't exist on WooPay.
* Before clicking on `Save my information for a faster and secure checkout` checkbox, fill out a phone number without the country code.
* Click on the `Save my information for a faster and secure checkout` checkbox.
* Do not change the phone number.
* Place Order.
* The user should be created on WooPay with the `+1` country code.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

File: client/components/platform-checkout/save-user/checkout-page-save-user.js [hunk 1]
```diff
@@ -61,6 +61,10 @@ const CheckoutPageSaveUser = ( { isBlocksCheckout } ) => {
 		// Take out any non-digit characters, except +.
 		phoneFieldValue = phoneFieldValue.replace( /[^\d+]*/g, '' );
 
+		if ( ! phoneFieldValue.startsWith( '+' ) ) {
+			phoneFieldValue = '+1' + phoneFieldValue;
```
Comment (frosso) [inline client/components/platform-checkout/save-user/checkout-page-save-user.js:67]: Can this truly be assumed? Is the prefix missing only for US phone numbers, or would it be missing for any phone number?

Comment (alefesouza) [reply client/components/platform-checkout/save-user/checkout-page-save-user.js:67]: The phone input on WooPay already converts numbers without country code to US by default [here](https://github.com/Automattic/woocommerce-payments/blob/8b3acaffac4892988726813ca36938e37a40c887/client/settings/phone-input/index.tsx#L75).

**Comment by pierorocca** [comment]
image

https://proccaatomic.wpcomstaging.com/

**Review by hsingyuc — COMMENTED** [review]
@alefesouza Sorry for the late review, and thank you for working on this!  In my environment, the user does get created on WooPay but the billing address, shipping address, and phone number are not saved.  Is it expected?  I can log in to WooPay as the user, but the country code doesn't seem to get added.

**Comment by hsingyuc** [comment]
I tested on `trunk`. The user gets created on WooPay but the billing address, shipping address, and phone number are also not saved. And I can't log in to WooPay with the user email.

**Comment by alefesouza** [comment]
Hi @hsingyuc, could you review this PR again against the latest changes? I've tested here and this bug still happens and is fixed with this PR.

**Comment by hsingyuc** [comment]
@alefesouza Does your environment save any information besides name and email?
Screen Shot 2023-03-15 at 8 15 16 PM

**Review by hsingyuc — COMMENTED** [review]
@alefesouza Does your environment save any information besides name and email?
Screen Shot 2023-03-15 at 8 15 16 PM

**Review by hsingyuc — APPROVED** [review]
I've checked the database `wp_woocommerce_address`, and the phone number does save in the correct format, so this is working well for me.

Do we intentionally not save the user billing and shipping address in the user meta table?  My billing fields are completely empty for the new user.

**Pull request merged**